### PR TITLE
Special searching for backward counterparts for circular routes

### PR DESCRIPTION
**Pull request by alexey-zakharenkov**
Solves #46.

Indeed, before Nov 27, 2023 the validator generated spurious notice _"Only one route in route_master. Please check if it needs a return route"_ for circular routes that started at different stations (like A-B-C-D-A and B-A-D-C-B) as if they had no return direction. After #44 the notice vanished, but it vanished at all for circular routes.

In this PR, the checking for return direction of circular routes is resumed, fixed and supplemented with tests.

### CODE REVIEW

**Review by mesozoic-drones — APPROVED** [review]

File: tests/test_route_master.py [hunk 1]
```diff
@@ -1,9 +1,51 @@
...
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [5, 6, 7],
+                "answer": [],
+            },
+            {
+                "sequence1": [1, 2, 3, 4],
+                "sequence2": [4, 3, 2, 1],
+                "answer": [1, 2],
+            },
```
Comment (SXHRYU) [inline tests/test_route_master.py:38]: I am not exactly sure how the function works, but here's more edge cases that I can think about just by looking at these ones:

1. `sequence1` is fully in `sequence2`;
2. `sequence1` = [], `sequence2` = [some numbers];
3. `sequence1` = [1, 2, 3, 4], `sequence2` = [4, 5, 6] (sequence2 starts from where sequence1 ends);

**Review by SXHRYU — CHANGES_REQUESTED** [review]

**Comment by SXHRYU** [comment]
And add `.venv` to .gitignore

File: subway_structure.py [hunk 2]
```diff
@@ -1617,15 +1623,253 @@ def find_twin_routes(self) -> dict[Route, Route]:
...
+                self.city.notice(
+                    "Should there be one stoparea or a transfer between "
+                    f"{st1.stoparea.station.name} {st1.stop} and "
+                    f"{st2.stoparea.station.name} {st2.stop}?",
+                    route1.element,
+                )
+
+    @staticmethod
+    def calculate_twin_routes_diff(route1: Route, route2: Route) -> tuple:
+        """Wagner–Fischer algorithm for stops diff in two twin routes."""
```
[the remark this answers is not in the document]
Comment (alexey-zakharenkov) [reply subway_structure.py:1810]: No, fastzy is not suitable in this case. Apart from it isn't generic to accept sequences other the character strings, it doesn't provide what we need - editorial prescription (steps to transform word1 to word2).

File: tests/test_route_master.py (hunk 1 above)
Comment (alexey-zakharenkov) [reply tests/test_route_master.py:68]: Tests added. Thanks.

**Review by SXHRYU — APPROVED** [review]
